providers/manage.rb

```diff
@@ -24,14 +24,44 @@ def whyrun_supported?
 use_inline_resources if defined?(use_inline_resources)
 
 action :create do
-  ssl_secret = Chef::EncryptedDataBagItem.load_secret(new_resource.data_bag_secret)
-  ssl_item =
-    begin
-      Chef::EncryptedDataBagItem.load(new_resource.data_bag, new_resource.search_id, ssl_secret)
-    rescue => e
-      raise e unless new_resource.ignore_missing
-      nil
-    end
+  # vault doesn't work in chef-solo
+  db_type = new_resource.data_bag_type
+  db_type = 'unencrypted' if db_type == 'vault' && Chef::Config[:solo]
+  case db_type
+  when 'encrypted'
+    ssl_secret = Chef::EncryptedDataBagItem.load_secret(new_resource.data_bag_secret)
+    ssl_item =
+      begin
+        Chef::EncryptedDataBagItem.load(new_resource.data_bag, new_resource.search_id, ssl_secret)
+      rescue => e
+        raise e unless new_resource.ignore_missing
+        nil
+      end
+  when 'unencrypted'
```

Comment:
An example would be storing a site wide CA public key for web browsers. This doesn't need to be encrypted and should be installed in the same places as the system CA certs. Most CA public key implementations work well with unencrypted data bag implementation. LDAP also requires clients to have the CA public key for encrypted binds to the server.